### PR TITLE
Add support for autoprefixer in compilescss

### PR DESCRIPTION
**Pull request by pegler**

### CODE REVIEW

**Comment by pegler** [comment]
hey @jrief - just a polite request for feedback on this. I'd love to get this included into the official repo rather than forking. We've been using this patch in production for almost a month without issue.

**Comment by AndreasBackx** [comment]
It might be best to add some tests for this as well.

**Comment by jrief** [comment]
@pegler sorry for the delay.

Are you also using django-compressor or django-pipeline?

During development, an autoprefixer is really handy, because it allows you to write native CSS code. However during production this belongs into tools such as django-compressor or django-pipeline, because there you normally do even more optimisation (prefixing, compression, uglyfication, etc.). Therefore, in this post-processing step, you have a pipe of Javascript commands.

Unless you neither use django-compressor nor django-pipeline – then that's a different story.

**Comment by pegler** [comment]
no problem at all @jrief!

We used to use django-compressor but have moved to single css file for our frontend, and greatly reduced our use of javascript, so we no longer needed it. We haven't looked into django-pipeline since django-sass-processor did almost everything we needed to compile our CSS. At first glance, it doesn't look like pipeline plays nicely with django-sass-processor and instead implements its own sass compilation. I also agree with django-compressor's statement [JS/CSS belong in the templates](https://django-compressor.readthedocs.io/en/latest/#why-another-static-file-combiner-for-django).

**Comment by jrief** [comment]
> We used to use django-compressor but have moved to single css file for our frontend, and greatly reduced our use of javascript, so we no longer needed it.

Ok, than that's a different story. I will try to reduce some of the code duplication, which your pull request would add to the project.

File: sass_processor/management/commands/compilescss.py [hunk 1]
```diff
@@ -284,6 +296,22 @@ def compile_sass(self, sass_filename, sass_fileurl):
         if self.sass_output_style:
             compile_kwargs['output_style'] = self.sass_output_style
         content = sass.compile(**compile_kwargs)
+        
+        # autoprefix CSS files using postcss in external JavaScript process
+        if self.node_npx_path and os.path.isdir(self.node_modules_dir or ''):
+            os.environ['NODE_PATH'] = self.node_modules_dir
+            try:
+                options = [node_npx_path, 'postcss', '--use', 'autoprefixer']
```
Comment (aszlig) [inline sass_processor/management/commands/compilescss.py]:
```suggestion
                options = [self.node_npx_path, 'postcss', '--use', 'autoprefixer']
```

**Comment by jrief** [comment]
Closing because Python-2 isn't an issue anymore.

**Pull request closed without merging**